app/pages/project/index.jsx

```diff
@@ -0,0 +1,428 @@
+counterpart.registerTranslations('nl', require('../../locales/nl').default);
+
+counterpart.setFallbackLocale('en');
+
+
+class ProjectPageController extends React.Component {
+  constructor() {
+    super();
+    this._listenedToPreferences = null;
+    this._boundForceUpdate = null;
+    this.state = {
+      background: null,
+      error: null,
+      guide: null,
+      guideIcons: {},
+      loading: false,
+      organization: null,
+      owner: null,
+      preferences: null,
+      project: null,
+      projectAvatar: null,
+      projectIsComplete: false,
+      projectRoles: null,
+      ready: false,
+      splits: null
+    };
+  }
+
+  componentWillMount() {
+    const { actions } = this.props;
+    if (this.props.location.query.language) {
+      actions.translations.setLocale(this.props.location.query.language);
+    }
+  }
+
+  componentDidMount() {
+    this._boundForceUpdate = this.forceUpdate.bind(this);
+    if (this.context.initialLoadComplete) { this.fetchProjectData(this.props.params.owner, this.props.params.name, this.props.user); }
+    this.setupSplits();
+  }
+
+  componentWillReceiveProps(nextProps, nextContext) {
+    const { owner, name } = nextProps.params;
+    const pathChanged = (owner !== this.props.params.owner) || (name !== this.props.params.name);
+    const userChanged = nextContext.initialLoadComplete && (nextProps.user !== this.props.user);
+    const initialLoadCompleted = nextContext.initialLoadComplete === !this.context.initialLoadComplete;
+
+    // Wait until we know if there's a user
+    if (pathChanged || userChanged || (initialLoadCompleted && (this.state.project === null))) {
+      if (!this.state.loading) { this.fetchProjectData(owner, name, nextProps.user); }
+      this.setupSplits(nextProps);
+    }
+  }
+
+  componentWillUnmount() {
+    Split.clear();
+  }
+
+  setupSplits(props) {
+    if (!props) {
+      ({ props } = this);
+    }
+    const { user } = props;
+    const { owner, name } = props.params;
+
+    if (user) {
+      Split.load(`${owner}/${name}`)
+      .then((splits) => {
+        this.setState({ splits });
+        if (!splits) {
+          return;
+        }
+        Object.keys(splits).forEach((split) => {
+          // log the first active split and break the loop
+          if (splits[split].state === 'active' && this.context.geordi) {
+            const experiment = splits[split].name;
+            const cohort = splits[split].variant ? splits[split].variant.name : undefined;
+            this.context.geordi.remember({ experiment, cohort });
+            return;
+          }
+        });
+        return;
+      });
+    } else {
+      Split.clear();
+      this.setState({ splits: null });
+      (this.context.geordi != null ? this.context.geordi.forget(['experiment', 'cohort']) : undefined);
+    }
+  }
+
+  getUserProjectPreferences(project, user) {
+    this.listenToPreferences(null);
+
+    const userPreferences = (user != null) ?
```

Comment:
I'm seeing explicit conditional checks against null all throughout the code. We don't typically write existential checks this way when writing new components and the decaffeinate comment at the top of the file suggests changing this if it makes sense. Perhaps a find and replace for all the `!= null` to remove?